content/blog/setting-up-harbor-as-a-local-container-registry-in-hpe-private-cloud-ai.md

```diff
@@ -0,0 +1,304 @@
+pcaidemo/cfe-nginx   v0.1.0    1e5f3c5b981a   2 months ago   192MB
+
+$ docker tag pcaidemo/cfe-nginx:v0.1.0 harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx:v0.1.0
+
+$ docker images
+REPOSITORY                                               TAG       IMAGE ID       CREATED        SIZE
+pcaidemo/cfe-nginx                                       v0.1.0    1e5f3c5b981a   2 months ago   192MB
+harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx   v0.1.0    1e5f3c5b981a   2 months ago   192MB
+```
+
+* *Push the image to Harbor registry*
+
+Push the image to the *Harbor* registry by running the following command:
+
+```shell
+$ docker push harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx:v0.1.0
+The push refers to repository [harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx]
+7e893c1b6ee8: Pushed
+463308bed0c9: Pushed
+4197a611afec: Pushed
+3e96162769d5: Pushed
+892e805f6f4f: Pushed
+626ab8a5d57b: Pushed
+7fb72a7d1a8e: Pushed
+v0.1.0: digest: sha256:114dff0fc8ee3d0200c3a12c60e3e2b79d0920dd953175ecb78a0b157425b25e size: 1778
+```
+
+* *Verify the image from Harbor registry* 
+
+From the *Harbor* UI, the image *cfe-nginx* appears under *Repositories* tab of the *demo* project:
+
+![](/img/demo-project.png)
+
+From  the Docker client, log in to the *Harbor* registry as the *pcai-developer* user, then pull the image *cfe-nginx* from the registry. The image downloads successfully, confirming that the user has appropriate access and the *Harbor* registry is functioning as expected.
+
+```shell
+$ docker login harbor.ingress.pcai0104.ld7.hpecolo.net
+Username: pcai-developer
+Password:
+
+WARNING! Your credentials are stored unencrypted in '/home/guoping/.docker/config.json'.
+Configure a credential helper to remove this warning. See
+https://docs.docker.com/go/credential-store/
+
+Login Succeeded
+
+$ docker images
+REPOSITORY   TAG       IMAGE ID   CREATED   SIZE
+$ docker pull harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx:v0.1.0
+v0.1.0: Pulling from demo/cfe-nginx
+b895f377d09e: Already exists
+3b00567da964: Pull complete
+56b81cfa547d: Pull complete
+1bc5dc8b475d: Pull complete
+979e6233a40a: Pull complete
+d2a7ba8dbfee: Pull complete
+32e44235e1d5: Pull complete
+Digest: sha256:114dff0fc8ee3d0200c3a12c60e3e2b79d0920dd953175ecb78a0b157425b25e
+Status: Downloaded newer image for harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx:v0.1.0
+harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx:v0.1.0
+
+$ docker images
+REPOSITORY                                               TAG       IMAGE ID       CREATED        SIZE
+harbor.ingress.pcai0104.ld7.hpecolo.net/demo/cfe-nginx   v0.1.0    1e5f3c5b981a   2 months ago   192MB
+```
+
+## Application deployment using Harbor registry
+
+With the container images pushed to the *Harbor* registry, the next step is to deploy the application to HPE PCAI using the same *Import Framework*, demonstrating how to pull images from *Harbor*. 
```

Comment:
I don't quite understand the last phrase in this sentence "demonstrating how to pull images from Harbor". It doesn't really flow with the first two parts of the sentence. Maybe this is a separate thought and requires its own sentence? For example, "It will demonstrate how to pull images from Harbor"?